screens/RegisterScreen.tsx

```diff
@@ -9,25 +9,26 @@ import {
   TouchableOpacity,
   LinkButton,
 } from "../components/Themed";
-import { emailRegisterPath } from "../constants/Api";
 import ButtonStyles from "../components/ButtonStyles";
-import { postRequest } from "../api/BaseRequests";
+import { ScreenWithNavigation } from "../types";
+import {
+  getAuth,
+  createUserWithEmailAndPassword,
+  sendEmailVerification,
+} from "firebase/auth";
 
-export default function RegisterScreen({ navigation }) {
+const auth = getAuth();
+export default function RegisterScreen({ navigation }: ScreenWithNavigation) {
   const [username, setUserName] = useState("");
   const [password, setPassword] = useState("");
   const [email, setEmail] = useState("");
 
   async function register() {
-    const body = {
-      username,
-      password,
-      email,
-    };
-    const resp = await postRequest(emailRegisterPath, body);
-    if(resp.success) {
-      navigation.navigate("Login");
-    }
+    createUserWithEmailAndPassword(auth, email, password).then(
+      (userCredential) => {
+        sendEmailVerification(userCredential.user);
+      }
+    );
```

Comment:
```suggestion
    const userCredential = await createUserWithEmailAndPassword(auth, email, password);
    sendEmailVerification(userCredential.user);
    );
```

Reply:
Doesn't this behave worse if registration fails? Tried to send email verification with undefined user?

Reply:
Good point, but I was thinking `createUserWithEmailAndPassword` would throw an error if it doesn't go through

Reply:
Gonna keep this for now because I think this syntax allows us to show an error toast easier in the future maybe?